dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl

```diff
@@ -216,6 +217,8 @@ RUN for iter in {1..10}; do \
     (exit $exit_code)
 
 {{- end }}
+# root group no more for elastic-agent user
+RUN gpasswd --delete {{ .user }} root
```

Comment:
my initial approach was to remove it there but apparently the npm flow is a little bit entangled with root and non-root and when I do remove it there I get the following our.

```
> [stage-1 15/17] RUN npm i -g --loglevel verbose --engine-strict @elastic/synthetics@stack_release &&     chmod ugo+rwX -R /usr/share/elastic-agent/.node:
0.479 npm verb exit -13
0.480 npm verb unfinished npm timer command:i 1718110923068
0.480 npm verb unfinished npm timer reify 1718110923070
0.480 npm verb unfinished npm timer reify:loadTrees 1718110923094
0.480 npm verb unfinished npm timer idealTree 1718110923094
0.480 npm verb unfinished npm timer idealTree:userRequests 1718110923097
0.480 npm verb code -13
0.480
0.480 npm ERR! Log files were not written due to an error writing to the directory: /usr/share/elastic-agent/.npm/_logs
0.480 npm ERR! You can rerun the command with `--loglevel=verbose` to see the logs in your terminal
------
Dockerfile:164
--------------------
 163 |     # If this fails dump the NPM logs
 164 | >>> RUN npm i -g --loglevel verbose --engine-strict @elastic/synthetics@stack_release && \
 165 | >>>     chmod ugo+rwX -R $NODE_PATH
 166 |     USER root
--------------------
ERROR: failed to solve: process "/bin/sh -c npm i -g --loglevel verbose --engine-strict @elastic/synthetics@stack_release &&     chmod ugo+rwX -R $NODE_PATH" did not complete successfully: exit code: 243
```

Thus I preferred to tackle this by removing it at the end 😄

Reply:
what do you think, should we leave this as is for this PR or should I try to untangle root vs rootless npm?

Reply:
NPM is only really use for the -complete image. It would be nice if it could be figured out. Downside of leaving it like this is it adds another layer to the container. Very small layer but still another layer.

Reply:
ok I sorted out the npm installation under elastic-agent user, the feel free to unresolve if you spot anything